src/core/jsproxy.c

```diff
@@ -1176,55 +1179,63 @@ JsMethod_cinit(PyObject* obj, JsRef this_)
 //
 // A subclass of JsProxy for Buffers
 
+// We make our own Buffer struct because as far as I can tell BytesArray and
+// array are both unsuitable. (To use "array" we need to perform extra copies,
+// using BytesArray we run into trouble finding a location to store the Shape.)
+// clang-format off
 typedef struct
 {
-  JsProxy super;
-  Py_ssize_t byteLength;
+  PyObject_HEAD
+  void* data;
+  Py_ssize_t byteLength; // invariant: byteLength should be equal to length * itemsize
+  Py_ssize_t length;
   char* format;
   Py_ssize_t itemsize;
-  PyObject* bytes;
-} JsBuffer;
+} Buffer;
```

Comment:
Well this buffer isn't actually backed by any Javascript data, it's actually just a general purpose blob of memory. This `Buffer` object is just here to patch a weird hole in the CPython API, namely that there is no way to say "create a new `PyObject` which is a C contiguous implementer of the Python Buffer protocol with this format and shape". (There are many Python and C functions that do stuff like this, but as far as I can tell all of them either leak memory or are very awkward to use.) 

Anyways since this is a general purpose buffer with no relationship to any Javascript object beyond the fact that it was created by copying a Javascript buffer, I figured `JsBuffer` was perhaps confusing. But either name is probably fine.

Reply:
Yeah so I want to reserve the name `JsBuffer` for a `JsProxy` of an ArrayBuffer view. This is a Python object with no direct connection to a `JsProxy` so it should have some different name.

Reply:
OK thanks!